src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green ' | 'blue';
 
+export interface Figure {
+  shape: Shape
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  color: Color;
+
+  constructor(
+    color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || b + c <= a || a + c <= b) {
+      throw new Error('Invalid triangle sides');
+    }
 
+    this.color = color;
```

Comment:
Checklist:
- When you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to do it manually.
![image](https://github.com/mate-academy/ts_figures_classes/assets/63243136/e5094947-8f5b-432a-9516-631f0427c3db)